elixir/lib/extension.ex

```diff
@@ -0,0 +1,23 @@
+defmodule SqliteVss.Extension do
+  @moduledoc """
+  Extension module for adding the sqlite-vss extension to an Ecto Repo.
+
+  In order to use this add the following to the `config.exs` file.
+
+  ```
+  config :my_app, MyApp.Repo,
+    after_connect: {SqliteVss.Extension, :enable, []}
+  ```
+  """
+
+  def enable(_conn) do
+    [db_conn] = Process.get(:"$callers")
+    db_connection_state = :sys.get_state(db_conn)
+    conn = db_connection_state.mod_state.state
```

Comment:
Could we just use the `_conn` parameter, or is this some elixir-ism?

Reply:
Great question!  The underscore means that we are going to ignore the argument.  The reason is that the connection being passed in is not the one we need to load the extension into.  I'm not exactly sure why we have different connections so I'll do a bit more research.